api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -43,8 +52,8 @@ public void close() {
       }
 
       @Override
-      public Iterator<E> iterator() {
-        return iterable.iterator();
+      public CloseableIterator<E> iterator() {
+        return CloseableIterator.withNoopClose(iterable.iterator());
```

Comment:
This changes the semantics of the iterator. Before, the iterators returned by this could be `Closeable`, but the `Iterable` itself was not `Closeable`. I think it would be unlikely to have an `Iterable` without a close operation that produced an `Iterator` that could be closed (since it should be a `CloseableGroup`) but I still think it makes sense to preserve whether the `Iterator` is `Closeable` here.

How about using `CloseableIterator.wrap` that calls `close` on the wrapped `Iterator` if it is `Closeable`?

Reply:
Actually, it looks like that's what `withClose` does. We should use that instead.